systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -10,64 +10,57 @@
 % the answer).
 %
 
-if model.use_new_kinsol
-  checkDirty(model);  
-  if nargin<6
+checkDirty(model);
+
+% method signature transition
+if nargin > 3
+  options_copy = options;
+end
+warn_signature_changed = false;
+if nargin == 5
+  warn_signature_changed = true;
+  options = struct();
+  options.compute_gradients = v;
```

Comment:
this points out the clear need to document the options structure at the top of the method.  :)

the dTdq, etc (and even the analytic jacobians) are clearly gradients which are computed right now, even when compute_gradients=false, so i think the name has to change. 
do all of these additional things that get build co-exist?  would it make sense to have multiple options for computing second derivatives, for computing geometry jac grads, etc.?  if a single option is still appropriate, then i'd be ok with even a `compute_additional_gradients` if it was well documented.